libcodechecker/analyze/analysis_manager.py

```diff
@@ -179,6 +179,19 @@ def init_worker(checked_num, action_num):
     progress_actions = action_num
 
 
+def get_dependent_headers(buildaction, archive):
+    LOG.debug("Generating dependent headers via compiler...")
+    try:
+        dependencies = set(create_dependencies(buildaction))
```

Comment:
Does it make sense to have this very thin wrapper that only calls `create_dependencies`? And maybe having `create_dependencies` return a set would be better than convert here?

Reply:
Actually, the purpose of this function is to handle any error which may happen during `create_dependencies`. Currently we just write the last exception string into the zip file, but on a long term it would be better to collect all the errors and pour them in once, that should be implemented in this function.